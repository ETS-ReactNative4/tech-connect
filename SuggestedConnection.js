import React, { Component } from 'react'
import NavigationService from './NavigationService.js';
import { StyleSheet, Text, View, Image, Button, TouchableHighlight } from 'react-native'
import Icon from 'react-native-vector-icons/Feather'

export default class SuggestedConnection extends Component {
  
  render() {
    const user = this.props.suggestion
    const { id, name, job_title, city } = this.props.suggestion
    return (
      <View style={styles.container}>
      <View style={styles.imageContainer}>
        <Image source={require('./profile-pic.jpeg')} style={styles.picture} />
      </View>
        <TouchableHighlight onPress={() => this.props.viewProfile(id)}>
          <Text style={ styles.name }>{ name }</Text>
        </TouchableHighlight>
        <Text style={ styles.position }>{ job_title }</Text>
        <View style={ styles.iconContainer }>
          <Icon name='map-pin' size={17} color='#93548F' style={ {marginRight: 2} } />
          <Text style={ {color: 'white', marginLeft: 3} }>{ city }</Text>
        </View>
        <TouchableHighlight style={ styles.connectBtn }>
          <Button title='Connect' color='white' onPress={() => NavigationService.navigate('Modal', {user})} />
        </TouchableHighlight>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#4AA9C5',
    height: 110,
    shadowOffset: {  width: 2,  height: 2 },
    shadowRadius: 4,
    shadowColor: 'black',
    shadowOpacity: .5,
    marginLeft: 50,
    marginRight: 40,
    borderTopLeftRadius: 6,
    borderTopRightRadius: 20,
    borderBottomRightRadius: 6,
    borderBottomLeftRadius: 6,
    marginBottom: 15,
  },
  imageContainer: {
    shadowOffset: {  width: 0,  height: 2 },
    shadowRadius: 10,
    shadowColor: 'black',
    shadowOpacity: .5,
  },
  picture: {
    height: 90,
    width: 90,
    borderRadius: 45,
    position: 'absolute',
    left: -40,
    top: 10
  },
  name: {
    marginLeft: 60,
    marginTop: 10,
    fontSize: 22,
    color: 'white'
  },
  position: {
    marginLeft: 60,
    marginTop: 2,
    fontSize: 17,
    color: 'white'
  },
  connectBtn: {
    position: 'absolute',
    right: -30,
    top: 35,
    backgroundColor: '#93548F',
    paddingRight: 10,
    paddingLeft: 10,
    borderRadius: 20,
    shadowOffset: {  width: 0,  height: 2 },
    shadowRadius: 10,
    shadowColor: 'black',
    shadowOpacity: .5,
  },
  iconContainer: {
    display: 'flex',
    flexDirection: 'row',
    marginLeft: 60,
    marginTop: 5
  },
})