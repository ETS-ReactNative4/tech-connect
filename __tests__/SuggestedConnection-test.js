import React from 'react'
import SuggestedConnection from '../SuggestedConnection'
import { TouchableHighlight } from 'react-native'
import { shallow } from 'enzyme'

describe('SuggestedConnection', () => {
  let wrapper 
  let mockSuggestion
  let mockViewProfile

  beforeEach(() => {
    mockSuggestion = {
      name: 'Howard',
      job_title: 'Astronomer',
      city: 'Denver',
      id: 10
    }
    mockViewProfile = jest.fn()
    wrapper = shallow(<SuggestedConnection 
      suggestion={ mockSuggestion } 
      viewProfile={ mockViewProfile } 
    />)
  })

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot()
  })

  it('should fire viewProfile on press of the name', () => {
    wrapper.find(TouchableHighlight).first().simulate('press')

    expect(mockViewProfile).toHaveBeenCalled()
  })
})