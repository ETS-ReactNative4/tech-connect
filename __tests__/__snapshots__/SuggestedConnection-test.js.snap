// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestedConnection should match the snapshot 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#4AA9C5",
      "borderBottomLeftRadius": 6,
      "borderBottomRightRadius": 6,
      "borderTopLeftRadius": 6,
      "borderTopRightRadius": 20,
      "height": 110,
      "marginBottom": 15,
      "marginLeft": 50,
      "marginRight": 40,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 2,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 4,
    }
  }
>
  <Component
    style={
      Object {
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 10,
      }
    }
  >
    <Component
      source={1}
      style={
        Object {
          "borderRadius": 45,
          "height": 90,
          "left": -40,
          "position": "absolute",
          "top": 10,
          "width": 90,
        }
      }
    />
  </Component>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onPress={[Function]}
    underlayColor="black"
  >
    <Component
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "marginLeft": 60,
          "marginTop": 10,
        }
      }
    >
      Howard
    </Component>
  </TouchableHighlight>
  <Component
    style={
      Object {
        "color": "white",
        "fontSize": 17,
        "marginLeft": 60,
        "marginTop": 2,
      }
    }
  >
    Astronomer
  </Component>
  <Component
    style={
      Object {
        "color": "white",
        "fontSize": 17,
        "marginLeft": 60,
        "marginTop": 2,
      }
    }
  >
    Denver
  </Component>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    style={
      Object {
        "backgroundColor": "#93548F",
        "borderRadius": 20,
        "paddingLeft": 10,
        "paddingRight": 10,
        "position": "absolute",
        "right": -30,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 10,
        "top": 35,
      }
    }
    underlayColor="black"
  >
    <Button
      color="white"
      title="Connect"
    />
  </TouchableHighlight>
</Component>
`;
